### PR TITLE
support see href and add seealso section for default template

### PR DESCRIPTION
**Pull request by ansyral**
#383

### CODE REVIEW

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs [hunk 1]
```diff
@@ -353,6 +353,33 @@ private IEnumerable<CrefInfo> GetMulitpleCrefInfo(XPathNavigator navigator, stri
...
+            foreach (XPathNavigator nav in iterator)
+            {
+                string altText = GetXmlValue(nav);
+                if (string.IsNullOrEmpty(altText)) altText = null;
+
+                string commentId = nav.GetAttribute("cref", string.Empty);
+                string url = nav.GetAttribute("href", string.Empty);
+                if (!string.IsNullOrEmpty(commentId))
+                {
+                    // Check if exception type is valid and trim prefix
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs]: remove `exception`

**Comment by superyyrrzz** [comment]
> </member>";

add some test case?

---
Refers to: test/Microsoft.DocAsCode.Metadata.ManagedReference.Tests/TripleSlashParserUnitTest.cs:92 in 5197fea. [](commit_id = 5197feaf4b2505fe59ab65091bc232213ba112a8, deletion_comment = False)

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutputUtility.cs [hunk 2]
```diff
@@ -94,5 +94,13 @@ public static string GetXref(string uid, string fullName = null, string name = n
             xref += "/>";
             return xref;
         }
+
+        public static string GetHref(string url, string altText = null)
+        {
+            var href = $"<span><a href=\"{HttpUtility.HtmlEncode(url)}\">";
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiBuildOutputUtility.cs]: no need to encode?

File: src/docfx.website.themes/default/partials/class.tmpl.partial [hunk 3]
```diff
@@ -197,3 +197,18 @@
...
 {{/extensionMethods}}
+{{#seealso.0}}
+<h3 id="seealso">{{__global.seealso}}</h3>
+<div class="seealso">
+{{/seealso.0}}
+{{#seealso}}
+    {{#isCref}}
+      <div>{{{type.specName.0.value}}}</div>
+    {{/isCref}}
+	{{^isCref}}
```
Comment (superyyrrzz) [inline src/docfx.website.themes/default/partials/class.tmpl.partial]: => 4 spaces

File: src/docfx.website.themes/common/common.js [hunk 4]
```diff
@@ -118,3 +119,30 @@ function getOverrideTemplate(uid) {
...
+        model.children.forEach(function (c) {
+            if (c.seealso) {
+                for(var key in c.seealso) {
+                    addIsCref(c.seealso[key]);
+                    seealso.push(c.seealso[key]);
+                }
+            }
+        });
+    }   
+    model.seealso = seealso || null;
```
Comment (superyyrrzz) [inline src/docfx.website.themes/common/common.js]: seeAlso

File: src/docfx.website.themes/common/common.js [hunk 5]
```diff
@@ -118,3 +119,30 @@ function getOverrideTemplate(uid) {
     content += "\n";
     return content;
 }
+
+function mergeSeeAlso(model) {
+    var seealso = [];
```
Comment (superyyrrzz) [inline src/docfx.website.themes/common/common.js]: addIsCref() already adds the key `isCref` to it. No need to construct a new list?

File: src/Microsoft.DocAsCode.DataContracts.ManagedReference/LinkInfo.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+{
+    using System;
+    using YamlDotNet.Serialization;
+
+    using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.Utility.EntityMergers;
+    using Newtonsoft.Json;
+
+    [Serializable]
+    public class LinkInfo
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.DataContracts.ManagedReference/LinkInfo.cs]: Minor: LinkInfo seems an extension of CrefInfo. Is it possible for exception to use this type with LinkType = CRef?

**Comment by vwxyzh** [comment]
split this commit?

File: src/docfx.website.themes/common/common.js (hunk 4 above)
Comment (ansyral) [inline src/docfx.website.themes/common/common.js]: it's "seealso" defined in itemviewmodel
```csharp
[YamlMember(Alias = "seealso")]
[JsonProperty("seealso")]
public List<LinkInfo> SeeAlsos { get; set; }
```

File: src/docfx.website.themes/common/common.js (hunk 5 above)
Comment (ansyral) [inline src/docfx.website.themes/common/common.js]: i'd like to merge children's seealso to parent's seealso section. namely, i think there should be only one SeeAlso section for a page.

File: src/Microsoft.DocAsCode.DataContracts.ManagedReference/LinkInfo.cs (hunk 6 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.DataContracts.ManagedReference/LinkInfo.cs]: seems exception only supports cref attribute, i think crefinfo is more appropriate?

**Comment by ansyral** [comment]
split to #459, #460

File: src/docfx.website.themes/common/common.js (hunk 5 above)
Comment (superyyrrzz) [inline src/docfx.website.themes/common/common.js]: Got your idea... should be a new list.

---
In reply to: [69418318](https://github.com/dotnet/docfx/pull/456#discussion_r69418318) [](ancestors = 69418318)

**Pull request closed without merging**